src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs

```diff
@@ -305,7 +305,7 @@ public void FinishedRecording(AVCaptureFileOutput captureOutput, NSUrl outputFil
 
 		public void SwitchFlash(CameraFlashMode newFlashMode)
 		{
-			if (isAvailable && device != null && newFlashMode != flashMode)
+			if (isAvailable && device != null && newFlashMode != flashMode && device.FlashAvailable)
```

Comment:
Shouldn't we move this check to the private ```void SwitchFlash()``` method?

I assume a crash appears inside that method and that method is invoking from different places, not only here.

Reply:
Actually, I do not understand why it crashes. There are checks which should prevent crashes + try-catch block
```if (device.IsFlashModeSupported(AVCaptureFlashMode.Off))```

Do you have a stack-trace?

Reply:
then the check should be in both places as this method first changes local variable and then goes to private method

Reply:
there are checks - yes, but the crash is during taking photo in other method. this happens becase - like I wrote above - it changes first local var and then checks. `TakePhoto()` uses local variable value and crashes because it has things set that can be illegal for some devices.

Reply:
I don't think it's a problem if we change the field's value (cuz it won't be used)

But my question is still relevant... Why does it even crash since checks exist?

Reply:
Well, then the code should be refactored a little more.
Cuz now you check only ```device.FlashAvailable``` 
But sometimes we should check ```device.TorchAvailable```. It depends on what value newFlashMode has.

Reply:
can one have torch without flash? interesting. let me check what I can do.

Reply:
> can one have torch without flash? interesting. let me check what I can do.
Not sure, but let's imaging another case (similar to yours) but when anyone trying to enable torch when it is not supported (flash is supported)
Your code will allow changing field's value, but then it will lead to a crash in TouchPhotoAsync (because Torch is not suppported).

Reply:
yes, that's true. I've added check for torch as well.